### PR TITLE
Port more cv tests

### PR DESCRIPTION
**Pull request by oshtaier**
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_check_composite_cv_addition_list_versions
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 17:47:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]

================================================================================= 1 passed in 165.82 seconds ====================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_publish_version_changes_in_source_env
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 17:51:05 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]

================================================================================= 1 passed in 136.56 seconds =================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_negative_add_same_package_filter_twice
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 17:55:44 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .
========================================================================== 1 passed, 32 warnings in 175.58 seconds ==========================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_delete_composite_version
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:01:38 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
=========================================================================== 1 passed, 4 warnings in 216.70 seconds ========================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_delete_non_default_version
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:06:22 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
=========================================================================== 1 passed, 4 warnings in 187.32 seconds ==========================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_delete_version_with_ak
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:11:37 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
=========================================================================== 1 passed, 1 warnings in 148.46 seconds ===================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_clone_within_diff_env
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:15:13 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]

================================================================================= 1 passed in 258.23 seconds =============================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_add_package_inclusion_filter_and_publish
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:19:59 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]

========================================================================== 1 passed, 10 warnings in 173.00 seconds =====================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_add_package_exclusion_filter_and_publish
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:24:18 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
========================================================================== 1 passed, 10 warnings in 178.54 seconds =======================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_update_exclusive_filter_package_version
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:27:47 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
========================================================================== 1 passed, 17 warnings in 240.70 seconds ========================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_promote_with_rh_custom_spin
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:32:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
========================================================================== 1 passed, 10 warnings in 298.50 seconds =================================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_add_errata_filter
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:38:33 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
=========================================================================== 1 passed, 6 warnings in 181.51 seconds =============================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_subscribe_system_with_custom_content
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:42:22 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]

================================================================================= 1 passed in 419.71 seconds ==========================================================
```
```
pytest tests/foreman/ui_airgun/test_contentview.py::test_positive_subscribe_system_with_rh_custom_spin
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-01-11 18:51:00 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_contentview.py .                                                                                                                                          [100%]
========================================================================== 1 passed, 10 warnings in 642.14 seconds ==========================================================
```

### CODE REVIEW

File: tests/foreman/ui/test_contentview.py [hunk 1]
```diff
@@ -644,641 +369,94 @@ def test_positive_update_name(self):
...
-                        'Failed to find content view %s from %s org' % (
-                            name, self.organization.name)
-                    )
-                    self.content_views.delete(name)
-
-    @run_in_one_thread
-    @run_only_on('sat')
-    @skip_if_not_set('fake_manifest')
-    @tier2
-    def test_positive_add_rh_custom_spin(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: That case has basically no sense neither in term of coverage (as 100% of it covered in `test_positive_edit_rh_custom_spin`) nor in term of logic

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -644,641 +369,94 @@ def test_positive_update_name(self):
...
-            status = self.content_views.promote(
-                cv_name, 'Version 1.0', env_name)
-            self.assertIn('Promoted to {}'.format(env_name), status)
-
-    @run_in_one_thread
-    @run_only_on('sat')
-    @skip_if_not_set('fake_manifest')
-    @upgrade
-    @tier3
-    def test_positive_publish_with_rh_custom_spin(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: it is redundant coverage as it is easy to track any issue with publish in similar test case on promote

File: tests/foreman/ui/test_contentview.py [hunk 3]
```diff
@@ -2053,171 +1231,6 @@ def test_negative_non_readonly_user_actions(self):
             # logout button
             self.browser.get(settings.server.get_url())
 
-    @run_only_on('sat')
-    @tier2
-    def test_positive_delete_default_version(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: `test_positive_remove_cv_version_from_default_env` basically has same test coverage

File: tests/foreman/ui/test_contentview.py [hunk 4]
```diff
@@ -2291,72 +1304,6 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
             status = self.content_views.promote(cv_name, 'Version 1', env_name)
             self.assertIn('Promoted to {}'.format(env_name), status)
 
-    @run_only_on('sat')
-    @tier2
-    def test_negative_create_puppet_env_without_module(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: covered in `test_positive_publish_with_force_puppet_env`

File: tests/foreman/ui/test_contentview.py [hunk 5]
```diff
@@ -2291,72 +1304,6 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
...
-                        name=cv_name,
-                        organization=self.organization,
-                    ).search()[0].id
-                ),
-            )
-            self.assertIsNone(self.environment.search(env_name))
-
-    @run_only_on('sat')
-    @tier2
-    def test_positive_create_puppet_env_without_module_and_force(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: covered in `test_positive_publish_with_force_puppet_env`

File: tests/foreman/ui_airgun/test_contentview.py [hunk 6]
```diff
@@ -1004,6 +1131,121 @@ def test_positive_remove_cv_version_from_default_env(session, module_org):
...
+
+
+@tier2
+def test_positive_delete_non_default_version(session):
+    """Delete a content-view version associated to non-default
+    environment
+
+    :id: 1c1beb36-e06b-419f-96db-43b4d85c5e25
+
+    :expectedresults: Deletion was plerformed successfully
```
Comment (pondrejk) [inline tests/foreman/ui_airgun/test_contentview.py]: nitpick "performed"

File: tests/foreman/ui/test_contentview.py [hunk 7]
```diff
@@ -644,641 +369,94 @@ def test_positive_update_name(self):
...
-                # assert the host_client exists in content hosts page
-                self.assertIsNotNone(
-                    self.contenthost.search(host_client.hostname))
+            for name in valid_data_list():
+                with self.subTest(name):
+                    make_contentview(
+                        session, org=self.organization.name, name=name)
+                    self.assertIsNotNone(
+                        self.content_views.search(name),
+                        'Failed to find content view %s from %s org' % (
```
Comment (pondrejk) [inline tests/foreman/ui/test_contentview.py]: Nitpick, I'd say new style formatting is preferable, several occurrences, up to you

File: tests/foreman/ui_airgun/test_contentview.py [hunk 8]
```diff
@@ -1102,6 +1389,82 @@ def test_positive_add_package_filter(session, module_org):
...
+            cv.name, VERSION, 'name = "{}"'.format(package1_name))
+        assert len(packages) == 1
+        assert packages[0]['Name'] == package1_name
+        packages = session.contentview.search_version_package(
+            cv.name, VERSION, 'name = "{}"'.format(package2_name))
+        assert not packages
+
+
+@tier2
+def test_positive_add_package_exclusion_filter_and_publish(session, module_org):
```
Comment (pondrejk) [inline tests/foreman/ui_airgun/test_contentview.py]: Isn't this also a candidate for removal? It seems like the scenario with exclude filter is also covered in test_positive_update_exclusive_filter_package_version

Comment (oshtaier) [reply tests/foreman/ui_airgun/test_contentview.py]: I thought about that, but we have slightly different scenario. We are working with different packages and different rule conditions, so I decided to leave both test cases

**Comment by oshtaier** [comment]
2 ACKs - merging

**Pull request merged**